### ISSUE TITLE
`reportUnusedVariable` doesn't report an unused variable

### ISSUE THREAD

**Issue by hamirmahal** [microsoft/pyright#10435]
**Describe the bug**

`reportUnusedVariable` doesn't report an unused variable, even with

```
[tool.pyright]
reportUnusedVariable = true
typeCheckingMode = "strict"
```

in `pyproject.toml`.

**Code or Screenshots**
```python
UNUSED_VARIABLE_1 = {"a": 1, "b": 2, "c": 3}


def add(a: int, b: int) -> int:
    """
    This function adds two numbers.

    >>> add(1, 2)
    3
    >>> add(2, 3)
    5
    """
    return a + b
```
**VS Code extension or command-line**
I'm running `pyright`

1. integrated into `Pylance`,
2. in a pre-commit check, and
3. in CI.

`pyright` passes in all 3.

```
$  pyright --version
pyright 1.1.400
```

**Comment by hamirmahal** [comment]
All checks pass at https://github.com/hamirmahal/reportUnusedVariable-does-not-report-unused-variable/commit/976733a9e5ac1d08a5d9bcbd9c1fe57b0f841d14.

**Comment by hamirmahal** [comment]
All checks pass at https://github.com/hamirmahal/reportUnusedVariable-does-not-report-unused-variable/commit/976733a9e5ac1d08a5d9bcbd9c1fe57b0f841d14, but I expected `pyright` to fail because `VARIABLE_11` is no longer used in `src/main.py`, or exported to any other file.

**Comment by hamirmahal** [comment]
This is similar, but not exactly the same as https://github.com/microsoft/pyright/issues/10102.

https://github.com/microsoft/pyright/issues/10102 mentions unused variables in a function definition.

This issue deals with unused variables at the top-level of a file.

**Comment by hamirmahal** [comment]
This is similar, but not exactly the same as https://github.com/microsoft/pyright/issues/10102.

https://github.com/microsoft/pyright/issues/10102 mentions unused variables in a function definition.

This issue deals with unused variables at the top level of a file.

**Comment by erictraut** [comment]
Pyright is working as intended here, so this isn't a bug.

A variable in the global scope is visible to other modules, so such variables are not considered unused. They can (and often are in practice) imported by other modules.

Python unfortunately has no mechanism for specifying which global (module-scoped) symbols are intended to be "exported" from the module. Most other modern languages do. In Python, all such variables are assumed to be exported.
